pkg/machines/components/cpuTypeModal.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from 'react';
+import cockpit from 'cockpit';
+import { Button, Form, FormGroup, FormSelect, FormSelectOption, FormSelectOptionGroup, Modal } from '@patternfly/react-core';
+
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import { setCpuMode as setCpuModeLibvirt } from '../../../libvirt-dbus.js';
+
+const _ = cockpit.gettext;
+
+export const CPUTypeModal = ({ vm, models, close }) => {
+    const [error, setError] = useState({});
+    const [cpuMode, setCpuMode] = useState(vm.cpu.mode);
+    const [cpuModel, setCpuModel] = useState(vm.cpu.model);
+
+    function save() {
+        setCpuModeLibvirt({
+            name: vm.name,
+            id: vm.id,
+            connectionName: vm.connectionName,
+            mode: cpuMode,
+            model: cpuModel
+        }).then(close, exc => setError({ dialogError: _("CPU configuration could not be saved"), dialogErrorDetail: exc.message }));
+    }
+
+    const defaultBody = (
+        <Form isHorizontal>
+            <FormGroup id="cpu-model-select-group" label={_("Mode")}>
+                <FormSelect value={cpuModel || cpuMode}
+                            aria-label={_("Mode")}
+                            onChange={value => {
+                                if ((value == "host-model" || value == "host-passthrough")) {
+                                    setCpuMode(value);
+                                    setCpuModel(undefined);
+                                } else {
+                                    setCpuModel(value);
+                                    setCpuMode("custom");
+                                }
+                            }}>
+                    <FormSelectOption key="host-model"
+                                      data-value="host-model"
+                                      value="host-model"
+                                      label="host-model" />
```

Comment:
shouldn't these labels be localized? I am wondering if they are static values, or they should be translatable (or we should have mapping from them to some human readable forms)?

Reply:
I am not sure if we want to localize these, these are attributes of the XML configuration and I think libvirt XML attributes should not be localized so that users can understand the mapping between our dialogs and libvirt XML files.
https://libvirt.org/formatdomain.html#cpu-model-and-topology
in the case the mapping it to:
``` xml
<cpu mode='host-model'>
```

``` xml
<cpu mode='host-passthrough'> 
```

Reply:
Fair enough.